Userland/Utilities/ls.cpp

```diff
@@ -195,16 +195,52 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     return status;
 }
 
+class ScopedPrinter {
+public:
+    ScopedPrinter(const char c)
+        : m_escape_character(c)
+    {
+        print();
+    }
+
+    ~ScopedPrinter()
+    {
+        print();
+    }
+
+    void print() const
+    {
+        if (m_escape_character)
+            out("{}", m_escape_character);
+    }
+
+private:
+    char m_escape_character { 0 };
+};
+
 static int print_escaped(StringView name)
```

Comment:
I strongly suggest using `Shell::escape_token` here instead (probably better to lift that function and its deps into a separate header, I think)
As it is, this doesn't really produce a good result, and it also doesn't really look like anything the shell would understand

e.g. it would "escape" `foo\"bar' .baz` as `'foo"bar' .baz'` - that's not quite right :P

Reply:
My implementation was indeed too naive, should be fixed now.